src/store/Store.js

```diff
@@ -27,6 +27,7 @@ class Store {
 
     this.extensionId = extensionId; // keep the extensionId as an instance variable
     this.port = chrome.runtime.connect(this.extensionId, {name: portName});
+    this.safetyMessage = chrome.runtime.onMessage.addListener(this.safetyHandler.bind(this));
```

Comment:
Instead of the safetyMessage approach, maybe we can just leverage [chrome.runtime.getBackgroundPage](https://developer.chrome.com/apps/runtime#method-getBackgroundPage). Once that resolves, we can then continue to do the port setup stuff. Thoughts? Seems like that might be simpler

Reply:
Cool.  I will play around with this idea today.  Avoiding additional messaging would be a boon for sure.

Reply:
🙌

Reply:
The main blocker I see with the `chrome.runtime.getBackgroundPage` approach is that it does not support content scripts.  So the store would fail in that instance.  I am gonna look around the API a bit more and see if there is anything else that might work.

Reply:
@collectedmind by "does not support content scripts", do you mean you can't call that from the content script?

Reply:
Sorry for the long turn around time, the weekend invaded.  To answer your question, yes.  You are not able to call that function from a content script.
  
![screen shot 2017-11-06 at 4 48 59 pm](https://user-images.githubusercontent.com/16119672/32471492-6e113532-c312-11e7-843f-b412b3b103a0.png)

Reply:
Regarding getBackgroundPage, the remark at the bottom of #115 might be relevant